src/odemis/driver/actuator.py

```diff
@@ -3550,3 +3550,160 @@ def stop(self, axes=None):
     def reference(self, axes):
         """Reference the linked axes stage"""
         raise NotImplementedError("Referencing is currently not implemented.")
+
+
+
+# "Sample Stage": {
+#     class: actuator.SampleStage,
+#     role: "stage",
+#     dependencies: {
+#         "under": "Stage"
+#     },
+#     affects: ["Camera", "EBeam"],
+#     metadata: {
+#         # Typically, x range is the same as FM_IMAGING_RANGE, and Y has to be converted
+#         POS_ACTIVE_RANGE: {"x": [0.040, 0.054], "y": [-30.e-3, 30.e-3]}  # TODO: migrate this to stage-bare? I think it's only used for tiled acq?
+#     },
+#     init: {},
+# }
+
+class SampleStage(model.Actuator):
+    """
+    Stage wrapper component which converts the stage position to the sample stage position.
+    The sample stage coordinates system is along the sample-plane which is adjusted
+    according to the pre-tilt and other factors.
+    """
+
+    def __init__(self, name, role, dependencies, posture_manager: MicroscopePostureManager, **kwargs):
+        """
+        dependencies (dict str -> actuator): name to objective lens actuator
+        """
+        if len(dependencies) != 1:
+            raise ValueError("SampleStage needs 1 dependency")
+
+        self._dependency = list(dependencies.values())[0]
+
+        model.Actuator.__init__(self, name, role, dependencies=dependencies,
+                                axes=copy.deepcopy(self._dependency.axes), **kwargs)
+
+        # posture manager, to convert the positions
+        # self.pm: MicroscopePostureManager = MicroscopePostureManager(model.getMicroscope())
+        self.pm = posture_manager
+
+        # RO, as to modify it the client must use .moveRel() or .moveAbs()
+        self.position = model.VigilantAttribute({"x": 0, "y": 0, "z": 0, "rx": 0, "rz": 0},
+                                                unit=("m", "m", "m", "rad", "rad"),  readonly=True)
+        # it's just a conversion from the dep's position
+        self._dependency.position.subscribe(self._updatePosition, init=True)
+
+        if model.hasVA(self._dependency, "referenced"):
+            self.referenced = model.VigilantAttribute({}, readonly=True)
+            self._dependency.referenced.subscribe(self._updateReferenced, init=True)
+
+        if model.hasVA(self._dependency, "speed"):
+            speed_axes = set(self._dependency.speed.value.keys())
+            if set(self.axes) <= speed_axes:
+                self.speed = model.VigilantAttribute({}, readonly=True)
+                self._dependency.speed.subscribe(self._updateSpeed, init=True)
+            else:
+                logging.info("Axes %s of dependency are missing from .speed, so not providing it",
+                             set(self.axes) - speed_axes)
+
+        # TODO: correctly modify the reference, speed and axes range attributes
+
+    # def _updateAxesRange(self):
+    #     """
+    #     Calculate the axes range of position VA from the given range of dep's.
+    #     """
```

Comment:
I don't think you'll need this, as normally the calibration values are not going to change live.